src/public/modules/forms/admin/components/response-components/response-attachment.client.component.js

```diff
@@ -8,27 +9,27 @@ angular.module('forms').component('responseAttachmentComponent', {
     encryptionKey: '<',
   },
   controllerAs: 'vm',
-  controller: [
-    'Submissions',
-    '$timeout',
-    responseAttachmentComponentController,
-  ],
+  controller: ['$q', '$timeout', responseAttachmentComponentController],
 })
 
-function responseAttachmentComponentController(Submissions, $timeout) {
+function responseAttachmentComponentController($q, $timeout) {
   const vm = this
 
   vm.downloadAndDecryptAttachment = function () {
     vm.hasDownloadError = false
-    Submissions.downloadAndDecryptAttachment(
-      vm.field.downloadUrl,
-      vm.encryptionKey.secretKey,
+    $q.when(
+      AdminSubmissionsService.downloadAndDecryptAttachment(
+        vm.field.downloadUrl,
+        vm.encryptionKey.secretKey,
+      ),
     )
       .then((bytesArray) => {
-        // Construct a downloadable link and click on it to download the file
-        let blob = new Blob([bytesArray])
-        // field.answer is the filename
-        triggerFileDownload(blob, vm.field.answer)
+        if (bytesArray) {
```

Comment:
previously, `Submissions.downloadAndDecryptAttachment` returned the result of decryption from the sdk. This means that in the old version, this could be `null` also. 

what then happens is that we call `triggerFileDownload`, which uses `FileSaver.saveAs` - this leads to a GET XMLHttpRequest being called on blob, which results in the blob potentially being saved (not too sure on this, the [docs](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open) assume a DOMString and this doesn't throw). 

however, if we don't pass in a name for the file, this leads to this - `name = name || blob.name || 'download'` where blob is what we passed in. because this isn't a try/catch, it crashes.

i think not doing anything on null is reasonable because there's no decrypted content - and hence, nothing to download!

Reply:
>i think not doing anything on null is reasonable

previously the crash would be caught in the `catch` block below and the user would see an error message, but now there's no feedback at all right?

Reply:
oh yea you're right; the `catch` block below sets a boolean variable which is then used by angular to determine what to render. good catch and i'll change it accordingly!